### PR TITLE
Added a ProxyFileHelper which currently creates a new file for images.

### PR DESCRIPTION
**Pull request by bisht-2400**

### CODE REVIEW

File: app/src/main/java/com/example/photogallery/Utils/FileHelper.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+public class FileHelper implements IFileHelper {
+    @Override
+    public File createFile(Activity context) throws IOException {
+        String timeStamp = new SimpleDateFormat("yyyyMMdd_HHmmss").format(new Date());
+        String imageFileName = "__" + timeStamp + "_";
+        File storageDir = context.getExternalFilesDir(Environment.DIRECTORY_PICTURES);
+        File image = File.createTempFile(imageFileName, ".jpg", storageDir);
+        return image;
+    }
+}
```
Comment (maennis) [inline app/src/main/java/com/example/photogallery/Utils/FileHelper.java:20]: Generally speaking, when the file format allows, it is considered best practice to add an empty newline at the end of each new file you introduce.  There are stylistic and functional reasons you can read about [here](https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log) if you are interested.  If we add linting to the project, I expect it will flag this.

File: app/src/main/java/com/example/photogallery/Presenters/GalleryPresenter.java [hunk 2]
```diff
@@ -53,13 +56,18 @@
...
     private IPhotoRepository photoRepository;
+    private IFileHelper fileHelper;
+
     private Activity context;
     private List<Photo> photos;
     private int index = 0;
 
     public GalleryPresenter(Activity context) {
         this.context = context;
+
```
Comment (maennis) [inline app/src/main/java/com/example/photogallery/Presenters/GalleryPresenter.java]: It is best practice to try to avoid adding unnecessary whitespace when you are committing  changes.  It starts to add extra noise for reviewers when they are going over your PR.

**Review by maennis — DISMISSED** [review]
I left a couple small comments about best practices, but the tests all passed and this looks fine to me.  You do have a conflict in the dev branch that needs to be resolved before we can merge.

**Review by maennis — CHANGES_REQUESTED** [review]
You need to resolve your merge conflict before.

**Comment by bisht-2400** [comment]
The Conflict seems to be solved.

**Review by maennis — APPROVED** [review]
Looks good to me
